ui/src/app/cohort-search/criteria-tree/node.component.ts

```diff
@@ -3,44 +3,33 @@ import {
   OnDestroy,
   OnInit,
   Input,
+  Output,
+  EventEmitter,
   ViewEncapsulation,
 } from '@angular/core';
 import {NgRedux, select} from '@angular-redux/store';
 import {Subscription} from 'rxjs/Subscription';
+import {List} from 'immutable';
 
 import {
   CohortSearchActions,
   CohortSearchState,
+  activeCriteriaList,
   isCriteriaLoading,
   criteriaChildren,
 } from '../redux';
 
 
 @Component({
   selector: 'app-criteria-tree-node',
-  template: `
-    <ng-container [clrLoading]="loading">
-      <clr-tree-node *ngFor="let node of children; trackBy: trackById">
-
-        <app-criteria-tree-node-info
-          [node]="node"
-          (onSelect)="handleSelection(node)">
-        </app-criteria-tree-node-info>
-
-        <ng-template clrIfExpanded *ngIf="node.get('group')">
-          <app-criteria-tree-node [node]="node">
-          </app-criteria-tree-node>
-        </ng-template>
-
-      </clr-tree-node>
-    </ng-container>
-  `,
+  templateUrl: './node.component.html',
 })
 export class CriteriaTreeNodeComponent implements OnInit, OnDestroy {
   @Input() node;
-
-  private children;
-  private loading;
+  @Output() isLoading = new EventEmitter<boolean>();
+  private _loading: boolean;
+  private children: List<any>;
+  private selections: List<any>;
```

Comment:
Again... should we be more specific on the typing?

Reply:
See above